otaclient/ota_proxy/ota_cache.py

```diff
@@ -533,16 +502,14 @@ def __init__(
         # NOTE 2: store unquoted URL in database
         self.meta = meta
         self._tracker = tracker  # bound tracker
-        self._executor = executor
         self._cache_commit_cb = commit_cache_callback
 
-    def _finalize_caching(self):
+    async def _finalize_caching(self):
         """Commit cache entry to db and rename tmp cached file with sha256hash
         to fialize the caching."""
-        self._cache_commit_cb(self.meta)
-        # for 0 size file, register the entry to db only,
-        # but if the 0 size file doesn't exist, create one
-        if self.meta.size > 0 or not (self._base_dir / self.meta.sha256hash).is_file():
+        await self._cache_commit_cb(self.meta)
+        # if the file with the same sha256has is already presented, skip the hardlink
+        if not (self._base_dir / self.meta.sha256hash).is_file():
```

Comment:
`self.meta.size > 0 or not` seems to be moved, intentional?

Reply:
Yes, and actually the previous one is wrong and will cause bug. 
The current one `if not (self._base_dir / self.meta.sha256hash).is_file()` is correct.